apollo-runtime/src/main/java/com/apollostack/android/ApolloConverterFactory.java

```diff
@@ -0,0 +1,34 @@
+package com.apollostack.android;
+
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.Moshi;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+
+import okhttp3.RequestBody;
+import okhttp3.ResponseBody;
+import retrofit2.Converter;
+import retrofit2.Retrofit;
+import retrofit2.converter.moshi.MoshiConverterFactory;
+
+public class ApolloConverterFactory extends Converter.Factory {
+  private final Moshi moshi;
+  private final MoshiConverterFactory moshiConverterFactory;
+
+  public ApolloConverterFactory(Moshi moshi) {
+    this.moshi = moshi;
+    moshiConverterFactory = MoshiConverterFactory.create(moshi);
+  }
+
+  @Override public Converter<ResponseBody, ?> responseBodyConverter(Type type, Annotation[] annotations,
+      Retrofit retrofit) {
+    JsonAdapter<?> adapter = moshi.adapter(type);
+    return new ApolloResponseBodyConverter<>(adapter);
```

Comment:
I guess we need to check if type doesn't subclass  of `GraphQLOperation.Data` then return null. 
So our adapter will be limited to parse only responses from graphql.

From java doc of retrofit: 
>/**
     * Returns a {@link Converter} for converting an HTTP response body to {@code type}, or null if
     * {@code type} cannot be handled by this factory. This is used to create converters for
     * response types such as {@code SimpleResponse} from a {@code Call<SimpleResponse>}
     * declaration.
     */